### PR TITLE
[1LP][RFR] General UI new tests

### PR DESCRIPTION
**Pull request by rlbabyuk**
Purpose or Intent
=================
General UI new tests
{{pytest: -vvv -k 'test_add_provider_trailing_whitespaces or test_configuration_large_number_of_tags'}}

### CODE REVIEW

File: cfme/tests/webui/test_general_ui.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    client.run_command('wget https://raw.githubusercontent.com/rhtconsulting/'
+                       'cfme-rhconsulting-scripts/master/rhconsulting_illegal_chars.rb')
+    client.run_command(
+        'cd /tmp && wget https://github.com/ManageIQ/manageiq/files/384909/tags.yml.gz'
+    )
+    client.run_command(
+        'gunzip tags.yml.gz && vmdb && bin/rake rhconsulting:tags:import[/tmp/tags.yml]'
+    )
+    category_groups = client.run_command('cat /tmp/tags.yml | grep description').\
+        output.split('\n- description:')
```
Comment (quarckster) [inline cfme/tests/webui/test_general_ui.py]: **REQUIRED**
```python
output = client.run_command('cat /tmp/tags.yml | grep description').output
category_groups = output.split('\n- description:')
```

File: cfme/tests/webui/test_general_ui.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+def test_configuration_large_number_of_tags(appliance, import_tags, soft_assert):
+    """Test page should be loaded within a minute with large number of tags"""
+    group = appliance.collections.groups.instantiate(description='EvmGroup-administrator')
+    view = navigate_to(group, 'Details')
+    for category, tags in import_tags.items():
+        for tag in tags:
+            soft_assert(view.entities.my_company_tags.tree.has_path(category.replace('  ', ' '),
+                                                                    tag.strip()), (
+                'Tag {} was not imported'.format(tag.strip())
+            ))
```
Comment (quarckster) [inline cfme/tests/webui/test_general_ui.py]: **REQUIRED**
```python
for category, tags in import_tags.items():
    category = category.replace('  ', ' ')
    for tag in tags:
        tag = tag.strip()
        soft_assert(view.entities.my_company_tags.tree.has_path(category, tag),
                   'Tag {} was not imported'.format(tag))
```

File: cfme/configure/access_control/__init__.py [hunk 3]
```diff
@@ -452,21 +471,18 @@ class GroupForm(ConfigurationView):
...
-        TAB_NAME = "My Company Tags"
-        tree_locator = 'tags_treebox'
-        tree = CbTree(tree_locator)
+    my_company_tags = View.nested(MyCompanyTags)
+    hosts_and_clusters = View.nested(Hosts_And_Clusters)
+    vms_and_templates = View.nested(Vms_And_Templates)
 
     @View.nested
-    class hosts_and_clusters(Tab):  # noqa
+    class Hosts_And_Clusters(Tab):  # noqa
```
Comment (mshriver) [inline cfme/configure/access_control/__init__.py]: @rlbabyuk This looks like these tab classes are duplicates, now that you've defined them above and used `View.nested(<viewclass>)` on L474, 475, 476. Are they serving some other purpose or just didn't get removed?

File: cfme/tests/webui/test_general_ui.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.markers.env_markers.provider import ONE_PER_TYPE
+from cfme.utils.appliance.implementations.ui import navigate_to
+
+
+@pytest.fixture()
+def import_tags(appliance):
+    client = appliance.ssh_client
+    client.run_command('cd /var/www/miq/vmdb/lib/tasks/')
```
Comment (mshriver) [inline cfme/tests/webui/test_general_ui.py]: I think we need some assertions in this chain of commands and output processing, to ensure that nothing is going wrong and cleanup+skip the test if so.

Consider asserting against each of these commands, so their RC is checked (if you expect 0 RC), and wrapping the whole block of commands in a `try/except` in order to catch the `AssertionError` and delete files + `pytest.skip()` with an explanation.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Pull request merged**